### PR TITLE
[TRTLLM-14198][refactor] Make FlashInfer a hard dependency for the Torch sampler

### PR DESCRIPTION
**Pull request by zhaoyangwang-nvidia**
…sampler and spec

Convert the Torch sampler and one-model speculative decoding from "FlashInfer-optional with torch fallback" to a hard FlashInfer dependency. The failure is raised at sampler / speculative-worker construction time (resolve_sampling_backend, SpecWorkerBase) via a new require_flashinfer helper and a sentinel-based bind_flashinfer_sampling, so sampling_utils stays importable for torch-only consumers (AutoDeploy demollm, drafting greedy helpers).

- Add require_flashinfer() in flashinfer_utils.py; IS_FLASHINFER_AVAILABLE semantics are unchanged (its ~46 legitimate fallback consumers are not touched).
- ops/flashinfer.py: drop the import guard, bind flashinfer.sampling at construction via a sentinel; compiled bodies do no import work.
- Remove the SimpleGroupedStrategySampler torch backend and the disable_flashinfer_sampling LlmArg + its plumbing; update the api_stability and usage golden files and the sampling/telemetry docs.
- compute_probs_from_logits becomes FlashInfer-only (drop the CUDA C++ op and CPU vanilla branches).
- speculative/interface.py: drop the availability guards, keep the flashinfer>=0.6.4 version gate and its version-gated branches.
- Delete now-dead fallback kernels (forward_native_sampling, _apply_top_k_top_p, _random_sample, vanilla.compute_probs_from_logits_op); keep beam-search/greedy/rejection/_Fusions and the torch sample() path used by rejection-sampling draft probs and demollm.
- Prune torch-vs-flashinfer parametrized sampler tests.

## Summary by CodeRabbit

* **New Features**
  * PyTorch sampling now consistently uses the FlashInfer-backed backend for supported workflows.

* **Bug Fixes**
  * Removed an unsupported sampling toggle from user-facing configuration and APIs.
  * Sampling-related behavior and logprob handling were aligned with the new default backend.

* **Documentation**
  * Updated developer and feature docs to reflect the simplified sampling configuration and current backend behavior.

* **Tests**
  * Updated sampling tests and API stability references to match the revised configuration surface.

## Description

Make FlashInfer required for the PyTorch-backend **Torch sampler**, and delete
the now-dead torch fallback paths and their tests. When flashinfer is missing,
`TorchSampler.__init__` now raises a clear `ImportError` instead of silently
running a slower/divergent torch path.

`IS_FLASHINFER_AVAILABLE` semantics are unchanged — its ~46 non-sampler
fallback consumers are not touched.

## What changed

- `TorchSampler.__init__` hard-fails (ImportError) when flashinfer is missing,
  before entering the CUDA-graph-captured sampling loop.
- Flatten the sampler backend: remove `SimpleGroupedStrategySampler` and the
  `GroupedStrategySampler` ABC; `FlashInferGroupedStrategySampler` is the only
  implementation. The one-line factory is inlined and `ops/interface.py` deleted.
- Remove the `disable_flashinfer_sampling` LlmArg and its plumbing (updates
  api_stability / usage golden files and sampling & telemetry docs).
- `compute_probs_from_logits` becomes FlashInfer-only; delete dead fallback
  kernels (`forward_native_sampling`, `_apply_top_k_top_p`, `_random_sample`,
  `vanilla.compute_probs_from_logits_op`).
- Prune torch-vs-flashinfer parametrized sampler tests.

## Intentionally kept (not fallbacks)

Torch `sample()` for two-model speculative rejection sampling and AutoDeploy
demollm; beam search (torch-only by design); `_Fusions` log-prob machinery;
and the `flashinfer>=0.6.4` version gate in one-model spec decoding.

## Test Coverage

Updated sampler unit tests (removed `use_flashinfer` / `disable_flashinfer_sampling`
parametrizations); updated api_stability references and usage golden manifest.

## PR Checklist

Please review the following before submitting your PR:
- PR description clearly explains what and why. If using CodeRabbit's summary, please make sure it makes sense.
- PR Follows [TRT-LLM CODING GUIDELINES](https://github.com/NVIDIA/TensorRT-LLM/blob/main/CODING_GUIDELINES.md) to the best of your knowledge.
- Test cases are provided for new code paths (see [test instructions](https://github.com/NVIDIA/TensorRT-LLM/tree/main/tests#1-how-does-the-ci-work))
- If PR introduces API changes, an appropriate PR label is added - either `api-compatible` or `api-breaking`. For `api-breaking`, include `BREAKING` in the PR title.
- Any new dependencies have been scanned for license and vulnerabilities
- [CODEOWNERS](https://github.com/NVIDIA/TensorRT-LLM/blob/main/.github/CODEOWNERS) updated if ownership changes
- Documentation updated as needed
- Update [tava architecture diagram](https://github.com/NVIDIA/TensorRT-LLM/blob/main/.github/tava_architecture_diagram.md) if there is a significant design change in PR.
- The reviewers assigned automatically/manually are appropriate for the PR.

- [x] Please check this after reviewing the above items as appropriate for this PR.

## GitHub Bot Help

To see a list of available CI bot commands, please comment `/bot help`.

### CODE REVIEW

File: tensorrt_llm/_torch/pyexecutor/sampler/sampling_utils.py [hunk 1]
```diff
@@ -968,12 +873,9 @@ def sampling_batch_spec_dec_one_model(
     is_greedy = temperatures <= vanilla._GREEDY_TEMPERATURE_THRESHOLD
     greedy_tokens = logits.argmax(dim=-1)
     logits = vanilla._safely_apply_temperature_inplace(logits, temperatures)
-    if IS_FLASHINFER_AVAILABLE:
-        sampled = flashinfer.top_k_top_p_sampling_from_logits_op(
-            logits, top_k, top_p, seed=seed, offset=offset
-        )
-    else:
-        sampled = vanilla.forward_native_sampling(logits, top_k, top_p)
+    sampled = flashinfer.top_k_top_p_sampling_from_logits_op(
```
Comment (QiJune) [inline tensorrt_llm/_torch/pyexecutor/sampler/sampling_utils.py:876]: This call is now unconditional, but SpecWorkerBase still sets use_flashinfer=False when FlashInfer is unavailable or below 0.6.4 (interface.py:901). The worker can therefore initialize successfully and fail here during the first non-greedy sampling step after the fallback was removed. Please replace that flag assignment with an availability/version check that raises ImportError in SpecWorkerBase.__init__.

Comment (zhaoyangwang-nvidia) [reply tensorrt_llm/_torch/pyexecutor/sampler/sampling_utils.py:876]: Thanks — fixed. SpecWorkerBase.__init__ now raises ImportError when flashinfer is unavailable or <0.6.4, instead of setting use_flashinfer=False and crashing later on the first non-greedy sampling step.

**Review by QiJune — APPROVED** [review]
LGTM

**Comment by zhaoyangwang-nvidia** [comment]
Hi @sunnyqgg Added some spec-side changes here touching the spec ↔ sampler integration — could you help review? Thanks!

**Review by sunnyqgg — APPROVED** [review]
LGTM
